sympy/series/sequences.py

```diff
@@ -0,0 +1,1144 @@
+from __future__ import print_function, division
+
+from sympy.core.expr import Expr
+from sympy.core.singleton import (S, Singleton)
+from sympy.core.symbol import Dummy
+from sympy.core.function import Lambda
+from sympy.core.add import Add
+from sympy.core.mul import Mul
+from sympy.core.compatibility import (range, integer_types, with_metaclass,\
+                                      is_sequence, iterable, ordered)
+from sympy.core.cache import cacheit
+from sympy.core.sympify import sympify
+from sympy.core.containers import Tuple
+from sympy.core.evaluate import global_evaluate
+from sympy.functions.elementary.integers import ceiling
+from sympy.functions.elementary.miscellaneous import Min, Max
+from sympy.polys import lcm
+from sympy.sets.sets import Interval, Set, Intersection
+from sympy.utilities.iterables import flatten
+
+
+################################################################################
+#                            SEQUENCES                                         #
+################################################################################
+
+
+class SeqBase(Expr):
+    """Base class for sequences"""
+
+    is_iterable = True
+
+    is_EmptySequence = False
+    is_SeqPer = False
+    is_SeqFunc = False
+    is_SeqFormula = False
+    is_SeqAdd = False
+    is_SeqMul = False
```

Comment:
Am 04.06.2015 um 19:35 schrieb Sartaj Singh:
>> +    is_iterable = True
>> +
>> +    is_EmptySequence = False
>> +    is_SeqPer = False
>> +    is_SeqFunc = False
>> +    is_SeqFormula = False
>> +    is_SeqAdd = False
>> +    is_SeqMul = False
>
> There is no greater reason. I find it convenient that's all. Should I remove these or it's fine?

Please remove.
They're redundant, and redundancy means that any changes in the future 
will have to be applied consistently. Imagine somebody refactoring stuff 
and suddenly seeing breakage because they didn't know how the `is_*` 
attributes are supposed to work (subclasses have a universal and 
predefined way how they work, so people will know what to do with them).

Nonredundant `is_*` attributes will have to stay, of course.